src/vivarium/framework/engine.py

```diff
@@ -262,29 +277,29 @@ def step(self) -> None:
                 self._clock.event_time,
             )
             self._logger.debug(f"Updating: {len(pop_to_update)}")
-            self.time_step_emitters[event](pop_to_update)
+            self.time_step_emitters[event](pop_to_update, None)
         self._clock.step_forward(self.get_population().index)
 
     def run(
         self,
         backup_path: Path | None = None,
-        backup_freq: int | float = None,
+        backup_freq: int | float | None = None,
     ) -> None:
-        if backup_freq:
+        if backup_freq and backup_path:
             time_to_save = time() + backup_freq
-            while self.current_time < self._clock.stop_time:
+            while self.current_time < self._clock.stop_time:  # type: ignore [operator]
```

Comment:
This an the next ignore are the same issues we have dealt with when comparing two ClockTime objects.